packages/js/src/plugins/candyMachineModule/operations/findCandyMachinesByPublicKeyField.ts

```diff
@@ -121,8 +122,45 @@ export const findCandyMachinesByPublicKeyFieldOperationHandler: OperationHandler
             ? parseCandyMachineCollectionAccount(unparsedCollectionAccount)
             : null;
 
-          return toCandyMachine(account, unparsedAccount, collectionAccount);
+          return {
+            account,
+            unparsedAccount,
+            collectionAccount,
+          };
         }
       );
+
+      // Find mint details for all unique SPL tokens used
+      // in candy machines that have non-null `tokenMint`
+      const tokenMints = new Set(
+        candyMachinesDetails
+          .map((details) => details.account.data.tokenMint)
+          .filter((tokenMint): tokenMint is PublicKey => tokenMint !== null)
+      );
+
+      let mints: Mint[] = [];
+
+      if (tokenMints.size > 0) {
+        const unparsedMintAccounts = await metaplex
+          .rpc()
+          .getMultipleAccounts([...tokenMints], commitment);
+
+        mints = unparsedMintAccounts.map((unparsedMintAccount) =>
+          toMint(toMintAccount(unparsedMintAccount))
+        );
+      }
+
+      // Then combine to return all `CandyMachine` results
+      return candyMachinesDetails.map((details) => {
+        const tokenMint = details.account.data.tokenMint;
+        return toCandyMachine(
+          details.account,
+          details.unparsedAccount,
+          details.collectionAccount,
+          tokenMint
+            ? mints.find((mint) => mint.address === tokenMint) ?? null
```

Comment:
You're comparing public keys by reference here. So the same public keys could still return false in this comparison. You need to use `mint.address.equals(tokenMint)`. Even better, instead of having a list of `Mint`, you could have a map `Record<string, Mint>` so finding a Mint is O(1).